addons/mail/wizard/mail_compose_message.py

```diff
@@ -452,8 +452,9 @@ def _process_recipient_values(self, mail_values_dict):
                     mail_to.append(mail_values['email_to'])
             # add email from recipients (res.partner)
             mail_to += [
-                recipient_emails[recipient_command[1]]
+                email
                 for recipient_command in mail_values.get('recipient_ids') or []
+                for email in email_re.findall(recipient_emails[recipient_command[1]])
```

Comment:
As it will have an impact on mass mailing, I would also add a test in ``test_mass_mailing/tests/test_mailing.py`` . You can look for example what we did with ``test_mailing_mailing_list_optout`` where we check the coherency between recipients and traces. Checking a bit how this change behaves with mass mailing app is important to be sure we don't do something unexpected :) .